crates/ra_ide/src/inlay_hints.rs

```diff
@@ -254,20 +255,57 @@ fn should_show_param_hint(
     parameters_len != 1 || !is_obvious_param(param_name)
 }
 
-fn is_argument_similar_to_param(argument: &ast::Expr, param_name: &str) -> bool {
-    let argument_string = remove_ref(argument.clone()).syntax().to_string();
-    let param_name = param_name.trim_start_matches('_');
-    let argument_string = argument_string.trim_start_matches('_');
-    argument_string.starts_with(&param_name) || argument_string.ends_with(&param_name)
+fn is_argument_similar_to_param_name(
+    sema: &Semantics<RootDatabase>,
+    argument: &ast::Expr,
+    param_name: &str,
+) -> bool {
+    if is_enum_name_similar_to_param_name(sema, argument, param_name) {
+        return true;
+    } else {
+        let argument_string = if let Some(repr) = get_string_representation(argument) {
+            repr
+        } else {
+            return false;
+        };
+        let argument_string = argument_string.trim_start_matches('_');
+        argument_string.starts_with(param_name) || argument_string.ends_with(param_name)
+    }
+}
+
+fn is_enum_name_similar_to_param_name(
+    sema: &Semantics<RootDatabase>,
+    argument: &ast::Expr,
+    param_name: &str,
+) -> bool {
+    match sema.type_of_expr(argument).and_then(|t| t.as_adt()) {
+        Some(Adt::Enum(e)) => &camel_case_to_snake_case(e.name(sema.db).to_string()) == param_name,
+        _ => false,
+    }
+}
+
+fn camel_case_to_snake_case(s: String) -> String {
```

Comment:
This is a copy-paste from `xtask`, `gen_syntax.rs`, `to_lower_snake_case` method and looks rather alien here, should I move it elsewhere?

Reply:
stdx ?

Reply:
We already indirectly depend on [heck](https://crates.io/crates/heck) through salsa, maybe we should just use that?

Reply:
I like stdx for now, but we can consider heck again if something else pops up.